src/main/java/teammates/ui/controller/InstructorFeedbackEditSaveAction.java

```diff
@@ -64,92 +65,7 @@ protected ActionResult execute() throws EntityDoesNotExistException {
-                    getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_PUBLISHTIME)));
-            break;
-        case Const.INSTRUCTOR_FEEDBACK_RESULTS_VISIBLE_TIME_ATVISIBLE:
-            newSession.setResultsVisibleFromTime(Const.TIME_REPRESENTS_FOLLOW_VISIBLE);
-            break;
-        case Const.INSTRUCTOR_FEEDBACK_RESULTS_VISIBLE_TIME_LATER:
-            newSession.setResultsVisibleFromTime(Const.TIME_REPRESENTS_LATER);
-            break;
-        case Const.INSTRUCTOR_FEEDBACK_RESULTS_VISIBLE_TIME_NEVER:
-            newSession.setResultsVisibleFromTime(Const.TIME_REPRESENTS_NEVER);
-            break;
-        default:
-            log.severe("Invalid resultsVisibleFrom setting editing " + newSession.getIdentificationString());
-            break;
-        }
-        
-        // handle session visible after results visible to avoid having a
-        // results visible date when session is private (session not visible)
-        type = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_SESSIONVISIBLEBUTTON);
-        switch (type) {
-        case Const.INSTRUCTOR_FEEDBACK_SESSION_VISIBLE_TIME_CUSTOM:
-            newSession.setSessionVisibleFromTime(TimeHelper.combineDateTime(
-                    getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_VISIBLEDATE),
-                    getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_VISIBLETIME)));
-            break;
-        case Const.INSTRUCTOR_FEEDBACK_SESSION_VISIBLE_TIME_ATOPEN:
-            newSession.setSessionVisibleFromTime(Const.TIME_REPRESENTS_FOLLOW_OPENING);
-            break;
-        case Const.INSTRUCTOR_FEEDBACK_SESSION_VISIBLE_TIME_NEVER:
-            newSession.setSessionVisibleFromTime(Const.TIME_REPRESENTS_NEVER);
-            // overwrite if private
-            newSession.setResultsVisibleFromTime(Const.TIME_REPRESENTS_NEVER);
-            newSession.setEndTime(null);
-            newSession.setFeedbackSessionType(FeedbackSessionType.PRIVATE);
-            break;
-        default:
-            log.severe("Invalid sessionVisibleFrom setting editing " + newSession.getIdentificationString());
-            break;
-        }
-        
-        String[] sendReminderEmailsArray = getRequestParamValues(Const.ParamsNames.FEEDBACK_SESSION_SENDREMINDEREMAIL);
-        List<String> sendReminderEmailsList = sendReminderEmailsArray == null ? new ArrayList<String>()
-                                                                              : Arrays.asList(sendReminderEmailsArray);
-        newSession.setOpeningEmailEnabled(sendReminderEmailsList.contains(EmailType.FEEDBACK_OPENING.toString()));
-        newSession.setClosingEmailEnabled(sendReminderEmailsList.contains(EmailType.FEEDBACK_CLOSING.toString()));
-        newSession.setPublishedEmailEnabled(sendReminderEmailsList.contains(EmailType.FEEDBACK_PUBLISHED.toString()));
-        
-        return newSession;
+        InstructorFeedbackAddAction feedbackData = new InstructorFeedbackAddAction();
```

Comment:
This dependency is not right. These two actions should be independent of each other.
If there are common methods, then these two actions should be inheriting an abstract class. Take a look at `InstructorCourseInstructorAbstractAction` and the two classes inheriting it for example.